ui/desktop/app/routes/scopes/scope/projects/targets/index.js

```diff
@@ -134,14 +116,30 @@ export default class ScopesScopeProjectsTargetsIndexRoute extends Route {
     if (orgScope.isOrg && scopes.length === 0) {
       query.filter = orgFilter;
     }
-    let targets = await this.store.query('target', query);
-    const totalItems = targets.meta?.totalItems;
+    const targetsPromise = this.store.query('target', query);
+
+    let [targets, sessions, allTargets] = await Promise.all([
+      targetsPromise,
+      sessionsPromise,
+      allTargetsPromise,
+      aliasPromise,
+    ]);
```

Comment:
I collected up all the promises to execute them in parallel rather than us doing them sequentially, it seemed like we were starting to have quite a few requests which adds up:

This is the time it takes for an already cached client daemon for a load with an all targets request and one without:
<img width="201" alt="Screenshot 2024-04-19 at 11 24 47 AM" src="https://github.com/hashicorp/boundary-ui/assets/5783847/f9948815-3c35-467b-a75f-135a0e8e9c3d">

And this is the same setup but with the promises running in parallel:
<img width="212" alt="Screenshot 2024-04-19 at 11 22 50 AM" src="https://github.com/hashicorp/boundary-ui/assets/5783847/70ee20b8-094a-49fb-9210-49763de4e636">